src/Core/Types.Sorting/QueryableSortVisitor.cs

```diff
@@ -129,5 +136,42 @@ IOrderedQueryable<TSource> sortedSource
         }
 
         #endregion
+
+        // Adapted from internal System.Web.Util.OrderingMethodFinder
+        // http://referencesource.microsoft.com/#System.Web/Util/OrderingMethodFinder.cs
+        class OrderingMethodFinder : ExpressionVisitor
+        {
+            bool _orderingMethodFound = false;
+
+            public override Expression Visit(Expression node)
```

Comment:
I changed the reference implementation slightly. This way it stops once a ordermethod is found